actuators/machine.go

```diff
@@ -45,116 +46,109 @@ const (
 type Machine struct {
 	Core       corev1.CoreV1Interface
 	ClusterAPI v1alpha1.ClusterV1alpha1Interface
-}
-
-// NewMachineActuator returns a new machine actuator object
-func NewMachineActuator(clusterapi v1alpha1.ClusterV1alpha1Interface, core corev1.CoreV1Interface) *Machine {
-	return &Machine{
-		Core:       core,
-		ClusterAPI: clusterapi,
-	}
+	Log        logr.Logger
 }
 
 // Create creates a machine for a given cluster
 // Note: have to print all the errors because cluster-api swallows them
 func (m *Machine) Create(ctx context.Context, c *clusterv1.Cluster, machine *clusterv1.Machine) error {
 	old := machine.DeepCopy()
-	fmt.Printf("Creating a machine for cluster %q\n", c.Name)
+	m.Log.Info("Creating a machine for cluster", "cluster-name", c.Name)
 	clusterExists, err := cluster.IsKnown(c.Name)
 	if err != nil {
-		fmt.Printf("%+v", err)
+		m.Log.Error(err, "Error finding cluster", "cluster", c.Name)
```

Comment:
This doesn't need the cluster name, but all of the places where m.Log.Error is replacing `fmt.Printf("%+v", err)` should include the stack trace in them, so this line needs to be

```
m.Log.Error(err, "Error finding cluster", "stacktrace", fmt.Sprintf("%+v", err))
```